lightning/src/ln/router.rs

```diff
@@ -1864,4 +1864,89 @@ mod tests {
 		assert!(router.should_request_full_sync(&node_id));
 		assert!(!router.should_request_full_sync(&node_id));
 	}
+
+	#[test]
+	fn handling_node_announcements() {
+		let (secp_ctx, _, router) = create_router();
+		let secret_key = &SecretKey::from_slice(&hex::decode("0202020202020202020202020202020202020202020202020202020202020202").unwrap()[..]).unwrap();
+		let node_id = PublicKey::from_secret_key(&secp_ctx, secret_key);
+		let zero_hash = Sha256dHash::hash(&[0; 32]);
+		let first_announcement_time = 500;
+
+		let mut unsigned_announcement = UnsignedNodeAnnouncement {
+			features: NodeFeatures::supported(),
+			timestamp: first_announcement_time + 1000,
+			node_id,
+			rgb: [0; 3],
+			alias: [0; 32],
+			addresses: Vec::new(),
+			excess_address_data: Vec::new(),
+			excess_data: Vec::new(),
+		};
+		let mut msghash = hash_to_message!(&Sha256dHash::hash(&unsigned_announcement.encode()[..])[..]);
+		let valid_announcement = NodeAnnouncement {
+			signature: secp_ctx.sign(&msghash, secret_key),
+			contents: unsigned_announcement.clone()
+		};
+
+	 	match router.handle_node_announcement(&valid_announcement) {
+			Ok(_) => panic!(),
+			Err(e) => assert_eq!("No existing channels for node_announcement", e.err)
+		};
+
+		{
+			let mut network = router.network_map.write().unwrap();
```

Comment:
In a few places, in general, tests should try to avoid modifying internal state directly unless there isn't really an easy way to do so. In this case, I think you can just as easily get away with not accessing the map directly and just handling announcements to add things, and check that they're there using the APIs to get announcements back out.